scripthost/src/main/kotlin/com/commonwealthrobotics/bowlerkernel/scripthost/Session.kt

```diff
@@ -0,0 +1,271 @@
+                                    msg.hasNewConfig() -> TODO("Not yet implemented")
+                                    else -> throw IllegalStateException("Unhandled message type: $msg")
+                                }
+
+                                logger.debug { "Result of handling client message: $result" }
+
+                                if (msg.hasRunRequest()) {
+                                    if (result is Either.Left) {
+                                        logger.warn(result.a) { "Script did not run successfully." }
+                                        send(
+                                            sessionServerMessage {
+                                                errorBuilder.requestId = msg.runRequest.requestId
+                                                errorBuilder.description =
+                                                    "Script did not run successfully: ${result.a.localizedMessage}"
+                                            }
+                                        )
+                                    }
+                                }
+
+                                true
+                            }
+                        }
+                    }
+                }
+            } while (cont)
+        } catch (t: Throwable) {
+            logger.error(t) { "Unhandled error in session" }
+            throw t
+        } finally {
+            getKoin().unloadModules(modules)
+            logger.debug { "Session ended" }
+        }
+    }.consumeAsFlow()
+
+    init {
+        require(coroutineScope != GlobalScope) {
+            "A session may not be started in GlobalScope."
+        }
+
+        @Suppress("TooGenericExceptionCaught") // Okay because we just log and rethrow
+        try {
+            getKoin().loadModules(modules)
+        } catch (t: Throwable) {
+            logger.error(t) { "Unhandled error when constructing session" }
+            throw t
+        }
+    }
+
+    /**
+     * Handles a [RunRequest].
+     *
+     * @param runRequest The [RunRequest]
+     * @param sessionChannel A channel to send messages to the client.
+     * @return The result of loading and running the script.
+     */
+    private suspend fun runRequest(
+        runRequest: RunRequest,
+        sessionChannel: SendChannel<SessionServerMessage>
+    ): Either<Throwable, Any?> {
+        val scriptLoader = getKoin().get<ScriptLoader>()
+
+        val script = sessionChannel.withTask(
+            runRequest.requestId,
+            nextTaskID.getAndIncrement(),
+            "Initializing ${runRequest.file.path}"
+        ) {
+            val script = scriptLoader.resolveAndLoad(runRequest.file, runRequest.devsList, runRequest.environmentMap)
+            scriptRequestMap[runRequest.requestId] = script
+            script
+        }
+
+        return sessionChannel.withTask(
+            runRequest.requestId,
```

Comment:
I didn't realize we'd have potentially multiple tasks created per request (when writing up the design doc I had figured that would be a 1:1 relationship). If that's the plan we're going with, I'd recommend clearly documenting that. The other option would be to consider the whole thing (i.e. loading and executing) as one "task" and update the description as it progresses.

Reply:
> If that's the plan we're going with

It is. Where do you want it to be documented?

Reply:
Adding onto this because I just remembered something, loading is logically (in my opinion) a sub-task. Would it make sense to change the `NewTask` proto so that tasks can be created from either a request or from a parent task? I think that's more in line with what we want. Thoughts?

Reply:
> It is. Where do you want it to be documented?

If we go with the 1:n model, then it should probably be documented in both the proto and the design doc (I can do this if you want). If we go with the sub-task model, that would need to be documented in the same places (and would require updating the proto itself).